### PR TITLE
move focus widget before deleting multi-instance

### PR DESCRIPTION
**Pull request by dterrahe**
This partially fixes one of the bugs that came up in #10384 (https://github.com/darktable-org/darktable/issues/10384#issuecomment-981770474)

Aparently there has been a long standing issue (concluding from the years old comment) that destroying the module header gui still leaves the focused widget focused. When the focus is then later on moved to something else, gtk wants to clean up the previously focussed one and crashes.

This fix forces the focus to the center widget whenever a multi-instance is deleted. I don't like doing this (because focus should stay where it is assigned and not move around on its own) but I haven't found a reliable way to determine if the module-to-be-deleted contained the focused widget or not; the obvious ways to do so failed. Which may also be why gtk itself doesn't take care of this correctly. The real underlying cause still escapes me, but I think this "workaround" is better than the previous one (which did not remove the module header at all!).

### CODE REVIEW

File: src/develop/imageop.c [hunk 1]
```diff
@@ -495,16 +495,9 @@ static void dt_iop_gui_delete_callback(GtkButton *button, dt_iop_module_t *modul
...
-    gtk_widget_hide(module->expander);
-
-    // we move the module far away, to avoid problems when reordering instance after that
-    // FIXME: ?????
-    gtk_box_reorder_child(dt_ui_get_container(darktable.gui->ui, DT_UI_CONTAINER_PANEL_RIGHT_CENTER),
-                          module->expander, -1);
-
     dt_iop_gui_cleanup_module(module);
-    gtk_widget_destroy(module->widget);
+    gtk_widget_grab_focus(dt_ui_center(darktable.gui->ui));
```
Comment (AlicVB) [inline src/develop/imageop.c:499]: that's the key to fix the issue

**Review by AlicVB — COMMENTED** [review]
looks good and works in my tests (I've done a quite similar fix)

That said, I wonder if we shouldn't split that in 2 PR in order to be more conservative considering that release is quite close now : 
1. with only the `gtk_widget_grab_focus` so that the issue is effectively fixed => 3.8
2. all the other change which seems ok to me but may have unwanted side effects => 4.0

**Comment by dterrahe** [comment]
I'm happy either way.

@TurboGit do you want me to submit this as 2 PRs or are you OK with all these changes or alternatively willing to just make the 1 liner change yourself and leave the rest for 4.0.

**Comment by TurboGit** [comment]
@dterrahe : What I would like is to have 2 commits in this PR. I'll merge both and if an issue if found before the release (we still have 3 weeks of field testing) we can easily and safely revert. Works for you?

**Review by TurboGit — APPROVED** [review]
Thanks!

**Pull request merged**